### PR TITLE
chore(rest): adding host nqn in node object

### PR DESCRIPTION
**Pull request by abhilashshetty04**
Node object in Rest will be reporting hostnqn in Spec and Status if set.

Signed-off-by: Abhilash Shetty <abhilash.shetty@datacore.com>

### CODE REVIEW

**Comment by abhilashshetty04** [comment]
bors try

File: control-plane/rest/openapi-specs/v0_api_spec.yaml [hunk 1]
```diff
@@ -2207,6 +2207,10 @@ components:
           description: the drain state
           allOf:
             - $ref: '#/components/schemas/CordonDrainState'
+        node_nqn:
+          description: hosts allowed to connect the target
```
Comment (tiagolobocastro) [inline control-plane/rest/openapi-specs/v0_api_spec.yaml]: Think you can remove the description and let it pick up the one from HostNqn automatically?

File: control-plane/rest/openapi-specs/v0_api_spec.yaml [hunk 2]
```diff
@@ -2207,6 +2207,10 @@ components:
...
+          allOf:
```
Comment (tiagolobocastro) [inline control-plane/rest/openapi-specs/v0_api_spec.yaml]: why do you need allOf here?

File: control-plane/rest/tests/v0_test.rs [hunk 3]
```diff
@@ -97,6 +97,8 @@ async fn client_test(cluster: &Cluster, auth: &bool) {
                 cluster.composer().container_ip(cluster.node(0).as_str())
             ),
             cordondrainstate: None,
+
```
Comment (tiagolobocastro) [inline control-plane/rest/tests/v0_test.rs]: remove empty line

File: common/src/types/v0/store/node.rs [hunk 4]
```diff
@@ -308,6 +308,7 @@ impl From<NodeSpec> for models::NodeSpec {
             src.endpoint.to_string(),
             src.id,
             src.cordon_drain_state.into_opt(),
+            src.node_nqn.map(|nqn| nqn.to_string()),
```
Comment (tiagolobocastro) [inline common/src/types/v0/store/node.rs]:
```suggestion
            src.node_nqn.into_opt(),
```

File: common/src/types/v0/transport/node.rs [hunk 5]
```diff
@@ -184,7 +184,12 @@ rpc_impl_string_id!(NodeId, "ID of a node");
 
 impl From<NodeState> for models::NodeState {
     fn from(src: NodeState) -> Self {
-        Self::new(src.grpc_endpoint.to_string(), src.id, src.status)
+        Self::new_all(
+            src.grpc_endpoint.to_string(),
+            src.id,
+            src.status,
+            src.node_nqn.map(|nqn| nqn.to_string()),
```
Comment (tiagolobocastro) [inline common/src/types/v0/transport/node.rs]:
```suggestion
            src.node_nqn.into_opt(),
```

File: control-plane/rest/openapi-specs/v0_api_spec.yaml (hunk 1 above)
Comment (abhilashshetty04) [reply control-plane/rest/openapi-specs/v0_api_spec.yaml]: Incorporated

File: control-plane/rest/tests/v0_test.rs (hunk 3 above)
Comment (abhilashshetty04) [reply control-plane/rest/tests/v0_test.rs]: Incorporated

File: common/src/types/v0/store/node.rs (hunk 4 above)
Comment (abhilashshetty04) [reply common/src/types/v0/store/node.rs]: Thanks. Incorporated

File: common/src/types/v0/transport/node.rs (hunk 5 above)
Comment (abhilashshetty04) [reply common/src/types/v0/transport/node.rs]: Incorporated

File: control-plane/rest/openapi-specs/v0_api_spec.yaml (hunk 2 above)
Comment (abhilashshetty04) [reply control-plane/rest/openapi-specs/v0_api_spec.yaml]: Not needed yes. Has only 1 schema. Removed allOf

**Review by tiagolobocastro — APPROVED** [review]

**Review by Abhinandan-Purkait — APPROVED** [review]

**Comment by abhilashshetty04** [comment]
bors merge

File: control-plane/rest/openapi-specs/v0_api_spec.yaml [hunk 6]
```diff
@@ -2207,6 +2207,8 @@ components:
           description: the drain state
           allOf:
             - $ref: '#/components/schemas/CordonDrainState'
+        node_nqn:
+          $ref: '#/components/schemas/HostNqn'
```
Comment (avishnu) [inline control-plane/rest/openapi-specs/v0_api_spec.yaml:2211]: Why is it HostNqn and not NodeNqn?

File: control-plane/rest/openapi-specs/v0_api_spec.yaml [hunk 7]
```diff
@@ -2226,6 +2228,8 @@ components:
           $ref: '#/components/schemas/NodeId'
         status:
           $ref: '#/components/schemas/NodeStatus'
+        node_nqn:
+          $ref: '#/components/schemas/HostNqn'
```
Comment (avishnu) [inline control-plane/rest/openapi-specs/v0_api_spec.yaml:2232]: Same comment as above.

Comment (tiagolobocastro) [reply control-plane/rest/openapi-specs/v0_api_spec.yaml:2232]: Hey @avishnu this has been merged now, but just for clarification, hostnqn is the actual nvme terminology.

Comment (avishnu) [reply control-plane/rest/openapi-specs/v0_api_spec.yaml:2232]: Ok, I was thinking about the naming consistency. There are "schemas/NodeId", "schemas/NodeStatus" and then there is "schemas/HostNqn", hence the comment.

Thanks !!

**Pull request merged**